Open Judge System/Workers/OJS.Workers.ExecutionStrategies/ExecutionStrategy.cs

```diff
@@ -19,14 +19,27 @@ public abstract class ExecutionStrategy : IExecutionStrategy
 
         public ExecutionResult SafeExecute(ExecutionContext executionContext)
         {
-            this.WorkingDirectory = DirectoryHelpers.CreateTempDirectory();        
+            Exception innerException = null;
+            this.WorkingDirectory = DirectoryHelpers.CreateTempDirectoryForExecutionStrategy();
             try
             {
                 return this.Execute(executionContext);
             }
+            catch (Exception e)
+            {
+                innerException = e;
```

Comment:
Have you tested upper exception propagation? What if the exception's inner exception has an inner exception and so forth? What gets logged in the end?

Reply:
Yes, all inner exceptions of this innerException are propagated to the logger